src/event.rs

```diff
@@ -354,6 +357,27 @@ pub enum WindowEvent<'a> {
     ThemeChanged(Theme),
 }
 
+/// Metadata which is passed along request to load clipboard.
+pub type ClipboardMetadata = dyn std::any::Any + Send + Sync + 'static;
+
+#[derive(Debug, Clone)]
+pub struct ClipboardContent {
+    /// Clipboard data.
+    pub data: Vec<u8>,
+
+    /// Mime type which was picked to be loaded.
+    pub mime: String,
+
+    /// Metadata passed to `request_clipboard_content`.
+    pub metadata: Option<std::sync::Arc<ClipboardMetadata>>,
+}
```

Comment:
It looks like there's no way to tell *which* clipboard request we're responding to. Is this supposed to be handled by the `ClipboardMetadata` in the consumer? I would have expected a serial.

Reply:
Yeah. However we in alacritty need to apply function on certain loads. We could use u64 and decide what to do based on u64 value. But I've decided to go for Any and downcasting.

Reply:
My preference here would be for `request_primary_clipboard_content` to return a serial.